monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import queue
+import threading
+from queue import Queue
+from threading import Event
+from typing import Callable, Dict, List
+
+from infection_monkey.i_puppet import IPuppet, PingScanData, PortScanData
+
+from .threading_utils import create_daemon_thread
+
+logger = logging.getLogger()
+
+Callback = Callable[[str, PingScanData, Dict[int, PortScanData]], None]
```

Comment:
Python is not a functional programming language, but maybe we can make it more readable? What does the first string stand for? `int` in `Dict[int, PortScanData]`?

Reply:
You'd like something more like this?

Reply:
You'd like something more like this?

```suggestion
IP = str
Port = int
Callback = Callable[[IP, PingScanData, Dict[Port, PortScanData]], None]
```

Reply:
Yes